### ISSUE TITLE
Compatibility with Night Owl VS Code theme

### ISSUE THREAD

**Issue by wooorm** [wooorm/markdown-tm-language#1]
_Originally reported by @kentcdodds at https://github.com/mdx-js/mdx-analyzer/issues/318_

### Steps to reproduce

Install the latest version of vscode-mdx
Open a markdown file with some markdown, even simple markdown will do. Here are some examples:

```md
# Hello world

This **should** be highlighted
```

<img width="396" alt="image" src="https://user-images.githubusercontent.com/1500684/227630902-2d73c0b4-207a-40ed-9280-7d1305723b55.png">

```md
import Comp from './comp'

<Comp prop={34} prop2="hi">
  This: `should be syntax highlighted`.
</Comp>
```

<img width="509" alt="image" src="https://user-images.githubusercontent.com/1500684/227631237-c1ca5510-fb41-4753-9eef-d7196f1694dd.png">

```md
## This should be highlighted

And _this_ should be too...
```

![image](https://user-images.githubusercontent.com/1500684/227631785-8b0f2c70-a168-44e6-bc15-defcc3803930.png)

```md
This is a list:

- why
- are
- these
- hyphens
- green?

No big deal, just kinda surprised...
```

![image](https://user-images.githubusercontent.com/1500684/227632025-322a20b4-3561-4af5-8e9f-8dd1bdfb14bb.png)

### Expected behavior

Things like headings, strong, italics, and inline code blocks should be highlighted.

### Actual behavior

The colors are either non-existent or different from what I was used to before. The components look great though 👍

I'm using Night Owl, but these behaviors show up in others as well. I guess what I would really like is if my MDX highlighting looked as close to the regular markdown highlighting as possible.

Here's what the examples above look like with regular markdown highlighting in Night Owl:

<img width="385" alt="image" src="https://user-images.githubusercontent.com/1500684/227633381-b05ee777-4abb-4295-acf8-2715942e3d1e.png">

<img width="494" alt="image" src="https://user-images.githubusercontent.com/1500684/227633441-32110b1c-35a4-44e2-9905-512f30c0cb1f.png">

(Yeah, MDX does better on this one for sure)

<img width="376" alt="image" src="https://user-images.githubusercontent.com/1500684/227633618-1dbc816f-7da9-435f-bca7-06c9320dbd53.png">

<img width="460" alt="image" src="https://user-images.githubusercontent.com/1500684/227633721-377d0247-6b97-4700-b810-2a9a313d17d0.png">

This could just be a case of "someone moved my cheese," but I do think there could be something wrong with the highlighting of the headings, strong, italics, and text in the back-tics.

Thanks!

**Comment by remcohaszing** [comment]
I agree. I also like to have inline markup fully highlighted, instead of just the wrapping tokens.

Headers are highlighted though. That might be theme related. Maybe it’s best to keep the token names as close to the VSCode markdown grammar as possible? (Assuming that one is the most widely used)

Also I like the highlighted list item markers.

**Comment by wooorm** [comment]
Hey Kent! :)

Thanks for the detailed report. And for the reference screenshots!

Important to know, is that the goal of this project is broad: to work in GitHub *and* in VS Code *and* in other places.
The most important is compatibility with GitHub, because we can’t change that. And that’s what **tons** of folks will see.
Then come VS Code themes, particularly the defaults first, then stuff like Night Owl!

Every tool and every theme will select different things and choose how to show them. So it’s always going to be different.

Here’s what I get with VS Code Solarized:

<img width="733" alt="Screenshot 2023-03-25 at 09 32 51" src="https://user-images.githubusercontent.com/944406/227706607-1d179f7a-122a-4ec4-8a53-4d076cb4614d.png">
<img width="818" alt="Screenshot 2023-03-25 at 09 33 09" src="https://user-images.githubusercontent.com/944406/227706611-4ac0a11f-2268-40bc-a4b2-456d0843cb78.png">
<img width="818" alt="Screenshot 2023-03-25 at 09 33 21" src="https://user-images.githubusercontent.com/944406/227706612-22123199-41d0-44d6-8b0c-d5fdbc9d71da.png">
<img width="812" alt="Screenshot 2023-03-25 at 09 33 36" src="https://user-images.githubusercontent.com/944406/227706615-8eea4f4f-5998-4edf-b531-900577f661b2.png">

So several things do work there, such as headings!

#### Headings

And that’s where we find a very important bug in Night Owl:

* Here’s [what we expose](https://github.com/wooorm/markdown-tm-language/blob/37c75fa6ffe72dc42c5a5a340feb431dc77616b8/grammar.yml#LL542C16-L542C16): `markup.heading.atx.1.mdx`
* Here’s [what VS Code Solarized selects](https://github.com/microsoft/vscode/blob/bf9141ba2358054beabd811b1c79bc46b5163bda/extensions/theme-solarized-dark/themes/solarized-dark-color-theme.json#L312): `markup.heading`
* Here’s [what Night Owl selects](https://github.com/sdras/night-owl-vscode-theme/blob/f260cea5356261337d18809bdbe922952c7a6fa4/themes/Night%20Owl-color-theme-noitalic.json#L1403): `markup.heading.markdown`

Looking at almost all scopes in Night Owl, they are super specific. Specific to JS, or to JSX, or to markdown.
And they’re not supposed to be like that, they’re supposed to as vague as possible (`markdup.heading`), unless you really need to differentiate `markup.heading.markdown` from others (e.g. `markup.heading.html`) or so.

So that’s where a lot of the problems come from!

#### Inline code

*   GitHub : `markup.raw`
*   VS Code Solarized: `markup.inline.raw`
*   VS Code Night Owl: `markup.inline.raw.markdown`

I think this is a weird choice in VS Code themes to go with `markup.inline.raw`.
`markup.raw.inline` makes more sense, then you can also have `markup.raw.block`.
But I’ve just pushed the name VS Code uses: 60fa64b.
Note that Night Owl still should drop the `.markdown` suffix.

#### Lists

Yep, the markers will typically be highlighted. Also the digits in ordered lists.
As for why it’s green for you?
The marker is tagged as `variable.unordered.list.mdx`.
This is [the name `atom/language-gfm` uses](https://github.com/atom/language-gfm/blame/ab4df763f6e28b7eadf1ac92ba5da5275fea33bf/grammars/gfm.cson#L1218).
I’m not sure why it’s a “variable”, but well, everyone supports it!
Including [Night Owl](https://github.com/sdras/night-owl-vscode-theme/blob/f260cea5356261337d18809bdbe922952c7a6fa4/themes/Night%20Owl-color-theme.json#L328-L334).

#### Emphasis (italics), strong (bold)

This is going to be a whole discussion, so I’ll open a separate issue for this!

---

I think the main take away here is that there’s a bug in Night Owl, it strictly specifies that only several scopes provided by [`vscode-markdown-tm-grammar`](https://github.com/microsoft/vscode-markdown-tm-grammar) get highlighted, which prevents this grammar from working.

**Comment by kentcdodds** [comment]
Thanks! I know it can be frustrating to have to triage issues that turn out not to be relevant for your own project, so thank you for taking the time to help me with my problem. I've opened an issue on Night Owl: https://github.com/sdras/night-owl-vscode-theme/issues/319

**Comment by wooorm** [comment]
No problem, thanks for raising an issue!
